### ISSUE TITLE
Unable to get available CDN servers from Steam!

### ISSUE THREAD

**Issue by ptable-1** [tpill90/steam-lancache-prefill#135]
The following content is from Google Translate.
When I follow the documentation to select-apps and then prefill, I get the error "Unable to get available CDN servers from Steam!".
So, I tried to comment out
Lines 44 to 46 of "SteamPrefill\Handlers\Steam\CdnPool.cs"
The code now looks like this
``` c#
          await _ansiConsole.StatusSpinner().StartAsync("Getting available CDNs", async _ =>
            {
                var retryCount = 0;
                while (_availableServerEndpoints.Count < _minimumServerCount && retryCount < _maxRetries)
                {
                    var allServers = await _steamSession.SteamContent.GetServersForSteamPipe();
                    _availableServerEndpoints.AddRange(allServers);

                    // Filtering out non-cacheable cdns, and duplicate hosts
                    // _availableServerEndpoints = _availableServerEndpoints.Where(e => e.Type == "SteamCache" && e.AllowedAppIds.Length == 0)
                    //                                                      .DistinctBy(e => e.Host)
                    //                                                      .ToList();

                    // Will wait increasingly longer periods when re-trying
                    retryCount++;
                    await Task.Delay(retryCount * 100);
                }
            });

            if (!_availableServerEndpoints.Any())
            {
                throw new CdnExhaustionException("Unable to get available CDN servers from Steam!");
            }
```

After I did this, everything worked fine. 
 So I doubt that the filtering of hosts is filtering out the cdn hosts that are available in my region.

I haven't learned C#, so I can't debug it further and modify the code.

**Comment by ptable-1** [comment]
I guess you will need logs from the lancache service,
[access.log](https://github.com/tpill90/steam-lancache-prefill/files/9582294/access.log)

**Comment by ptable-1** [comment]
I tried to debug the program and now I get the variable allServer in vscode
Maybe this will be useful to solve the problem
![屏幕截图 2022-09-19 165826](https://user-images.githubusercontent.com/31264658/190984417-cc68ef15-d0de-4ce9-8308-12ec66d27874.png)

**Comment by tpill90** [comment]
Released a fix for this in v1.5.2.  Please feel free to reopen this issue if the problem still persists.

**Comment by C9Glax** [comment]
![grafik](https://user-images.githubusercontent.com/13404778/193474352-5872803d-8ca1-40b0-af6e-a563250a3991.png)
Bug still exists in 1.5.2
After 2-3 retries it works.

**Comment by C9Glax** [comment]
[here](https://github.com/tpill90/steam-lancache-prefill/blob/master/SteamPrefill/Handlers/Steam/CdnPool.cs)
```
public async Task PopulateAvailableServersAsync()
 {
[...]
// Filtering out non-cacheable cdns, and duplicate hosts
_availableServerEndpoints = _availableServerEndpoints.Where(e => e.Type == "SteamCache" && e.AllowedAppIds.Length == 0)
  .DistinctBy(e => e.Host)
  .ToList();
[...]
});
```
It might help to pass `cellId` in the request to increase download-speeds [reference](https://github.com/SteamRE/SteamKit/blob/111054ec16747ef6bc03137b41f4fc3e185a9619/SteamKit2/SteamKit2/Steam/Handlers/SteamContent/SteamContent.cs#L22) and not just get NA Servers.
Although I haven't seen anything official on which ID correlates to what region, I would roughly say:
```
1-3, 6, 10-13 NA
4 EU-London
5, 7, 15 EU-Frankfurt
9 SEA
14 EU-Paris
8, 15+ are CDN
```

Also if  you take a look at the results of `_steamSession.SteamContent.GetServersForSteamPipe()` [click](https://steamapi.xpaw.me/#IContentServerDirectoryService/GetServersForSteamPipe)
At least for me not once the value AllowedAppIds is listed.
If my guess is correct it would be GameId of Steam?
Maybe get rid of this part for now, or filter for the actual AppId if Length > 0?

Other than that I am coming up short. It would be helpful to print (or store) some information (for example .json response) in case of failure...

**Comment by tpill90** [comment]
Hi @C9Glax ,  thank you for the detailed report + debugging + PR!  I really appreciate you digging into this deeper,  I will certainly start reviewing the changes you've proposed as soon as I get a few spare minutes to do so!

One other thing I was wondering, do you have Discord?  It might be helpful for us to communicate over chat tpill90#5322

**Comment by tpill90** [comment]
This issue should be resolved with #166, and was released in [v1.6.0](https://github.com/tpill90/steam-lancache-prefill/releases/tag/v1.6.0)

Please feel free to re-open this issue if the problem persists
